ctru-rs/examples/camera-image.rs

```diff
@@ -0,0 +1,205 @@
+
+    gfx.flush_buffers();
+    gfx.swap_buffers();
+    gfx.wait_for_vblank();
+
+    println!("\nPress R to take a new picture");
+    println!("Press Start to exit to Homebrew Launcher");
+
+    while apt.main_loop() {
+        hid.scan_input();
+        keys_down = hid.keys_down();
+
+        if keys_down.contains(KeyPad::KEY_START) {
+            break;
+        }
+
+        if keys_down.bits() & KeyPad::KEY_R.bits() != 0 {
+            println!("Capturing new image");
+            gfx.flush_buffers();
+            gfx.swap_buffers();
+            gfx.wait_for_vblank();
+            take_picture(&mut cam, &mut buf);
+        }
+
+        let img = convert_image_to_rgb8(buf, 0, 0, WIDTH as usize, HEIGHT as usize);
+
+        unsafe {
+            gfx.top_screen
+                .borrow_mut()
+                .get_raw_framebuffer(Side::Left)
+                .ptr
+                .copy_from(img.as_ptr(), img.len());
+        }
+
+        gfx.flush_buffers();
+        gfx.swap_buffers();
+        gfx.wait_for_vblank();
+    }
+}
+
+fn take_picture(cam: &mut Cam, buf: &mut [u8]) {
+    let buf_size = cam
+        .get_max_bytes(WIDTH, HEIGHT)
+        .expect("Failed to get max bytes");
+
+    cam.set_transfer_bytes(CamPort::PORT_BOTH, buf_size, WIDTH, HEIGHT)
+        .expect("Failed to set transfer bytes");
+
+    cam.activate(CamSelect::SELECT_OUT1_OUT2)
+        .expect("Failed to activate camera");
+
+    cam.clear_buffer(CamPort::PORT_BOTH)
+        .expect("Failed to clear buffer");
+    cam.synchronize_vsync_timing(CamSelect::SELECT_OUT1, CamSelect::SELECT_OUT2)
+        .expect("Failed to sync vsync timings");
+
+    cam.start_capture(CamPort::PORT_BOTH)
+        .expect("Failed to start capture");
+
+    let receive_event = cam
+        .set_receiving(buf, CamPort::PORT_CAM1, SCREEN_SIZE, buf_size as i16)
+        .expect("Failed to set receiving");
+
+    let receive_event2 = cam
+        .set_receiving(
+            &mut buf[SCREEN_SIZE as usize..],
+            CamPort::PORT_CAM2,
+            SCREEN_SIZE,
+            buf_size as i16,
+        )
+        .expect("Failed to set receiving");
+
+    unsafe {
+        let mut r = ctru_sys::svcWaitSynchronization(receive_event, WAIT_TIMEOUT);
+        if r < 0 {
+            panic!("Failed to wait for handle synchronization");
+        }
+        r = ctru_sys::svcWaitSynchronization(receive_event2, WAIT_TIMEOUT);
+        if r < 0 {
+            panic!("Failed to wait for handle 2 synchronization");
+        }
+    };
+
+    cam.play_shutter_sound(CamShutterSoundType::SHUTTER_SOUND_TYPE_NORMAL)
+        .expect("Failed to play shutter sound");
+
+    unsafe {
+        let mut r = ctru_sys::svcCloseHandle(receive_event);
+        if r < 0 {
+            panic!("Failed to close handle");
+        }
+        r = ctru_sys::svcCloseHandle(receive_event2);
+        if r < 0 {
+            panic!("Failed to close handle 2");
+        }
+    };
+
+    cam.activate(CamSelect::SELECT_NONE)
```

Comment:
Looking at the documentation, it's clear the different combinations are just bitwise OR versions. We can ditch the unions and just have 3 variants:
`Inward`
`Outward1`
`Outward2` (maybe if we understood which side is which we could call them `OutwardLeft` and `OutwardRight`).

These three variants are the only public ones available via our module, and thus the only one usable with `.activate()` (using bitwise ORs to include more than one).
After that, we can have `.deactivate()` which just sets it to `SELECT_NONE` under the hood.

Reply:
Outward1 seems to the right camera (facing outward when taking a picture).

Also did testing and it seems the camera combinations can be activated individually and don't have to be passed through to the libctru function at the same time. The reason I thought otherwise was because when deactivating it is passing 0 which would "overwrite" the current active camera. 

That being said, I am going to make functions specific for each. I will do the same for the shutter sounds.

More things to look into:

When using one camera is it always set to `CamPort::CAM1`?
When using multiple are the ports set in the order they are activated?

Reply:
The `IN1_OUT1` and other combination seems useless. Not sure if I'm missing something or it wasn't fully implemented or working in libctru to begin with. Need to test more with activating both outward to make sure it activates both.

Reply:
> The `IN1_OUT1` and other combination seems useless. Not sure if I'm missing something or it wasn't fully implemented or working in libctru to begin with. Need to test more with activating both outward to make sure it activates both.

In all honesty I have no idea how it could be using the inward and outward camera’s at the same time either, so that’s something strange.

Reply:
Here are my findings:

Activating `IN1` deactivates `OUT1` and/or `OUT2`. (Similar to `NONE` deactivating the other options)
`CamPort::CAM1` always refers to `OUT1` when combined or whichever camera is activated when only one is.

With this in mind, what are your thoughts on a struct/module for each that manages each variation? 

`Inward1`
`OutwardLeft`
`OutwardRight`
`Outward3D` (for managing `OUT1` and `OUT2` and their ports combined)

Reply:
The enum variants look fine, and it depends on whether we have a separate function for each variation or not.

Regarding the cam selection for actually taking the pictures:
We should only really thinking about the case with both outward cameras, right? In any other case it will just use CAM1, so we can abstract it by just having a “take_picture” method in `Cam`.

When it comes to take two buffers, it is slightly different, but maybe we could pass an enum option (which holds either one or two Vec buffers). This means we would have to handle the memory allocation inside Cam, rather than passing a reference to a pre-built buffer, but I don’t know if that’s a bad idea or not.